src/resolve/resolved-dns-rr.c

```diff
@@ -826,6 +827,115 @@ static char *format_txt(DnsTxtItem *first) {
         return s;
 }
 
+static char *format_svc_param_value(DnsSvcParam *i) {
+        char *value = NULL;
+        int r;
+
+        switch (i->key) {
+        case DNS_SVC_PARAM_KEY_ALPN: {
+                size_t sz = 0, offset = 0;
+                _cleanup_strv_free_ char **values_strv = NULL;
+                while (offset < i->length) {
+                        char *alpn;
+                        sz = (uint8_t) i->value[offset++];
+                        alpn = strndup((char *)&i->value[offset], sz);
+                        if (!alpn)
+                                return NULL;
+                        strv_push(&values_strv, alpn);
+                        offset += sz;
+                }
+                value = strv_join(values_strv, ",");
```

Comment:
oom check

Reply:
Well I actually just return `value` from here after the break, so it seems like its actually the other oom checks on strv_join in the other case:'s that are superfluous. I don't mind just adding one here for consistency and clarity though I guess.